h2o-core/src/main/java/water/util/CompressionFactory.java

```diff
@@ -0,0 +1,57 @@
+package water.util;
+
+import water.Iced;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.zip.GZIPOutputStream;
+
+public class CompressionFactory extends Iced<CompressionFactory> {
+
+  private String _name;
+
+  private CompressionFactory(String name) {
+    _name = name;
```

Comment:
Enum would provide us more safety here and would make sure the object is properly constructed, before failing later during a method call.

Reply:
just curious: why? it is a private constructor and the `make` method prevents from returning an invalid instance; IMHO it is functionally equivalent